### PR TITLE
fix: publish conditions

### PR DESCRIPTION
**Pull request by mkolasinski-splunk**
Fix the issue where automated releases where not created due to incorrect conditions for pre-publish job.

Tests:
PR scenario: https://github.com/splunk/test-addonfactory-repo/actions/runs/11027730555
release scenario: https://github.com/splunk/test-addonfactory-repo/actions/runs/11028553116
custom version release: https://github.com/splunk/test-addonfactory-repo/actions/runs/11047251558

### CODE REVIEW

**Review by mbruzda-splunk — DISMISSED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+### Checklist
+
+- [ ] `README.md` has been updated or is not required
+- [ ] push trigger tests
+- [ ] manual release test
+- [ ] automated releaes test
+- [ ] pull request trigger tests
+- [ ] schedule trigger tests
+- [ ] workflow errors/warnings reviewed and addressed
```
Comment (kdoroszko-splunk) [inline .github/pull_request_template.md:13]: May be we could be more direct here: 
```suggestion
- [ ] workflow errors/warnings reviewed and addressed

### Testing done 
(for each selected checkbox, the corresponding test results link should be listed here)
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+### Description
+
+(PR description goes here)
+
+### Checklist
+
+- [ ] `README.md` has been updated or is not required
+- [ ] push trigger tests
```
Comment (kdoroszko-splunk) [inline .github/pull_request_template.md:8]: push to `main`, `develop` or both?

Comment (kdoroszko-splunk) [reply .github/pull_request_template.md:8]: And also: we want that tests executed on that repo: [test-addonfactory-repo](https://github.com/splunk/test-addonfactory-repo) ?

File: .github/workflows/reusable-build-test-release.yml [hunk 3]
```diff
@@ -2377,8 +2377,9 @@ jobs:
 
   publish:
     if: |
-      (!cancelled() && needs.pre-publish.result == 'success' && github.event_name != 'pull_request' && github.event_name != 'schedule') ||
+      (!cancelled() && needs.pre-publish.result == 'success' && github.event_name == 'push' && github.event.inputs.custom-version == '') ||
```
Comment (kdoroszko-splunk) [inline .github/workflows/reusable-build-test-release.yml:2380]: This needs to be verified, but could we simplify the condition to be just: `!cancelled() && needs.pre-publish.result == 'success' && (github.event_name == 'push' || needs.validate-custom-version.result == 'success')`

**Review by kdoroszko-splunk — APPROVED** [review]

**Pull request merged**